src/compute/cuda/cuda_device.cpp

```diff
@@ -48,26 +48,31 @@ cuda_device::cuda_device(int device)
 {
 }
 
+int cuda_device::get_index() const noexcept
+{
+    return m_device;
+}
+
 std::unique_ptr<device_queue> cuda_device::create_queue()
 {
-    return std::make_unique<cuda_device_queue>(m_device);
+    return std::make_unique<cuda_device_queue>(*this);
```

Comment:
Why `*this`? Shouldn't it be `this::get_index()`?

Reply:
Now the constructor of `cuda_device_queue` accepts the `cuda_device` itself (and internally calls `get_index`)